src/main/java/org/broadinstitute/hellbender/utils/diffengine/DiffElement.java

```diff
@@ -0,0 +1,173 @@
+            final String subtree = value.substring(1, value.length()-1);
+            final DiffNode rec = DiffNode.empty(name, parent);
+            final String[] subParts = subtree.split(" ");
+            for ( final String subPart : subParts ) {
+                rec.add(fromString(subPart, rec.getBinding()));
+            }
+            return rec.getBinding();
+        } else {
+            return new DiffValue(name, parent, value).getBinding();
+        }
+    }
+
+    /**
+     * Makes a new element from parsing the tree. Puts the new element right under the ROOT.
+     */
+    public static DiffElement fromString(final String tree) {
+        return fromString(tree, DiffElement.ROOT);
+    }
+    /**
+     * Returns the name of this element.
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Returns the parent element of this element.
+     */
+    public DiffElement getParent() {
+        return parent;
+    }
+
+    /**
+     * Returns the value of this element.
+     */
+    public DiffValue getValue() {
+        return value;
+    }
+
+    /**
+     * Returns whether this element is the ROOT.
+     */
+    public boolean isRoot() { return this == ROOT; }
+
+    @Override
+    public String toString() {
+        return getName() + "=" + getValue().toString();
+    }
+
+    public String toString(final int offset) {
+        return (offset > 0 ? Strings.repeat(" ", offset) : 0) + getName() + "=" + getValue().toString(offset);
+    }
+
+    /**
+     * Returns the name of this element prefixed by the name of all of its parents.
+     */
+    public String fullyQualifiedName() {
+        if ( isRoot() ) {
+            return "";
+        } else if ( parent.isRoot() ) {
+            return name;
+        } else {
+            return parent.fullyQualifiedName() + "." + name;
+        }
+    }
+
+    /**
+     * Returns a one-line representation of this element.
+     */
+    public String toOneLineString() {
+        return getName() + "=" + getValue().toOneLineString();
+    }
+
+    /**
+     * If this element's value is a DiffNode, this method returns that DiffNode.
+     * Otherwise, it throws a GATKException.
+     */
+    public DiffNode getValueAsNode() {
+        if ( getValue().isCompound() ) {
+            return (DiffNode) getValue();
+        } else {
+            throw new GATKException("Illegal request conversion of a DiffValue into a DiffNode: " + this);
+        }
+    }
+
+    /**
+     * Returns the size of this element, which is 1 + the size of the element's value.
+     */
+    public int size() {
+        return 1 + getValue().size();
+    }
+
+    /**
+     * Collect differences between this element and the test element.
+     */
+    public List<Difference> diffElements(final DiffElement test) {
+        final DiffValue masterValue = this.getValue();
+        final DiffValue testValue = test.getValue();
+
+        if ( masterValue.isCompound() && masterValue.isCompound() ) {
```

Comment:
This looks broken:

        if ( masterValue.isCompound() && masterValue.isCompound() ) {

Should it perhaps read:

        if ( masterValue.isCompound() && testValue.isCompound() ) {

instead?